cloud/scope/common.go

```diff
@@ -23,3 +28,22 @@ func createLinodeClient(apiKey string) *linodego.Client {
 
 	return &linodeClient
 }
+
+func getCredentialDataFromRef(ctx context.Context, crClient client.Client, credentialsRef *corev1.SecretReference) ([]byte, error) {
+	secretRefName := client.ObjectKey{
+		Name:      credentialsRef.Name,
+		Namespace: credentialsRef.Namespace,
+	}
+
+	credSecret := &corev1.Secret{}
+	if err := crClient.Get(ctx, secretRefName, credSecret); err != nil {
+		return nil, fmt.Errorf("getting credentials secret %s\\%s: %w", secretRefName.Namespace, secretRefName.Name, err)
+	}
+
+	rawData, ok := credSecret.Data["apiToken"]
```

Comment:
Do we want the key to be configurable? For flexibility, we can use `corev1.SecretKeySelector` which allows for specifying a secret `name` and `key` (this assumes we expect the Secret to exist in the same namespace).

Reply:
I think it could, but I wanted to keep the key consistent with the key for the "global" API token we use throughout controller right now.

I can make it configurable if we want to start generalizing out this logic. I was thinking about removing `LINODE_TOKEN` as an environment variable completely (in a separate PR 😅) and just having the reconciller(s) reference Secrets directly.

Reply:
Understood! If there's precedence for using `apiToken` I don't see why we shouldn't stick with it and document it. And +1 on removing the environment variable as well.

Reply:
That's fair! I can add the `apiToken` key to the documentation 🫡